airflow/executors/celery_executor.py

```diff
@@ -341,11 +341,13 @@ def _check_for_stalled_adopted_tasks(self):
         just resend them. We do that by clearing the state and letting the
         normal scheduler loop deal with that
         """
-        now = utcnow()
+        sorted_adopted_task_timeouts = OrderedDict(
+            sorted(self.adopted_task_timeouts.items(), key=lambda k: k[1])
+        )
```

Comment:
Probably don’t even need `OrderedDict` since all we do is calling `items()` later anyway.

This should be good enough:

```
sorted_adopted_task_timeouts = sorted(self.adopted_task_timeouts.items(), key=lambda k: k[1])

for key, stalled_after in sorted_adopted_task_timeouts:
    ...
```

Also I feel having `utcnow()` outside of the loop is slightly better? It saves some function calls, unless millisecond accuracy is actually important here. But that’s minor either way.

Reply:
You're right